### PR TITLE
Add integration testing to smart_contract_upgrade

### LINKED ISSUE #264: Add integration tests to the `smart_contract_upgrade` example
**Task description**

Add integration tests to the `smart_contract_upgrade` example once the new integration testing library is released.

### PR DESCRIPTION
**Pull request by DOBEN**
## Purpose

close #264 

## Changes

Add integration testing to the `smart_contract_upgrade` example. The focus is on testing the upgrade with/without the migration function.

### CODE REVIEW

File: examples/smart-contract-upgrade/contract-version1/tests/tests.rs [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+            receive_name: OwnedReceiveName::new_unchecked("smart_contract_upgrade.upgrade".into()), // The receive function to call.
+            message: OwnedParameter::from_serial(&input_parameter)
+                .expect("`UpgradeParams` should be a valid inut parameter"), // The parameter sent to the contract.
+            amount: Amount::from_ccd(0), // Sending the contract 0 CCD.
+        },
+    );
+
+    assert_eq!(
+        update.expect("Upgrade should succeed").state_changed,
+        false,
```
Comment (abizjak) [inline examples/smart-contract-upgrade/contract-version1/tests/tests.rs:95]:
```suggestion
    assert!(
        !update.expect("Upgrade should succeed").state_changed
```

File: examples/smart-contract-upgrade/contract-version1/tests/tests.rs [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+            receive_name: OwnedReceiveName::new_unchecked("smart_contract_upgrade.upgrade".into()), // The receive function to call.
+            message: OwnedParameter::from_serial(&input_parameter)
+                .expect("`UpgradeParams` should be a valid inut parameter"), // The parameter sent to the contract.
+            amount: Amount::from_ccd(0), // Sending the contract 0 CCD.
+        },
+    );
+
+    assert_eq!(
+        update.expect("Upgrade should succeed").state_changed,
+        true,
```
Comment (abizjak) [inline examples/smart-contract-upgrade/contract-version1/tests/tests.rs:167]:
```suggestion
    assert!(
        update.expect("Upgrade should succeed").state_changed,
```

**Review by abizjak — COMMENTED** [review]
The test looks good, but I think we should always try to have our examples build with the version of concordium-std in the repository.

Can you please look into that using the suggestion in the comments?

**Review by Bargsteen — COMMENTED** [review]
Looks good! It's nice to see it being used ;) 
But yes, we should use the local `concordium-std` if possible.

**Review by Bargsteen — APPROVED** [review]

File: examples/smart-contract-upgrade/contract-version1/Cargo.toml [hunk 3]
```diff
@@ -16,9 +16,16 @@ wee_alloc = ["concordium-std/wee_alloc"]
...
 concordium-std = {path = "../../../concordium-std", default-features = false}
 
+[dev-dependencies]
+concordium-smart-contract-testing = "1.0"
+
 [lib]
 crate-type=["cdylib", "rlib"]
 
+[patch.crates-io]
+concordium-contracts-common = {path = "../../../concordium-contracts-common/concordium-contracts-common"}
```
Comment (abizjak) [inline examples/smart-contract-upgrade/contract-version1/Cargo.toml:26]: Can you just adda  comment as to why we do this please?

**Pull request merged**